main.go

```diff
@@ -132,30 +132,33 @@ func main() {
 
 	// If the executable is temporary, copy it to the full path, then restart
 	if strings.Contains(path, "-temp") {
-		err := copyExe(path, updater.BinPath(path))
+		newPath := updater.BinPath(path)
+		err := copyExe(path, newPath)
 		if err != nil {
+			log.Println("Copy error: ", err)
 			panic(err)
 		}
 
-		Systray.Restart()
+		Systray.Update(newPath)
```

Comment:
`Restart()` does not pass the path of the binary to start
`Update()` sets this path and then calls `Restart()`

Reply:
See the implementation of [`Update`](https://github.com/arduino/arduino-create-agent/blob/8ecf2dab58ae08a010d61bdaf00dfeac6a71b981/systray/systray.go#L73-L76) and [`Restart`](https://github.com/arduino/arduino-create-agent/blob/8ecf2dab58ae08a010d61bdaf00dfeac6a71b981/systray/systray.go#L27-L59)